apps/test-limit-switch/src/main.cpp

```diff
@@ -0,0 +1,30 @@
+#include "mbed.h"
+#include "PinNames.h"
+#include "LimitSwitch.h"
+
+DigitalIn Pin(PA_0);
+GPIO::LimitSwitch hSwitch(Pin);
+GPIO::LimitSwitch lSwitch(Pin, 0);
+
+int main() {
+  // Check to see if limit switch is pressed
+  while (true) {
```

Comment:
`while (true)` loops are infinite, so as of now, only the first one will run. You should only have one `while (true)` loop and put all your checks inside it.

Reply:
Thanks, I'll adjust that